### ISSUE TITLE
The future of the Ruby Toolbox

### ISSUE THREAD

**Issue by colszowka** [rubytoolbox/rubytoolbox#1]
Hello everyone,

as you might have heard, last week the server hosting the Ruby Toolbox crashed. While it's definetely possible to re-build from backups, I would like to use this opportunity to set in motion a discussion about the future of the site.

Launched in 2009, the Toolbox has always been a closed-source site. There's a bunch of reasons for that, but the most notable one was that I did not want to take on the burden of having to manage an open source community around it, having seen how much of a time-sink maintaining SimpleCov has been for a long time. This view has changed over the past few years as a bunch of smart and handsome people joined the simplecov team as contributors. I also believe the general attitude of open source users has shifted, caused by a whole bunch of "open-source-burnout" blog posts to a more positive, and encouraging community, replacing a wide-spread feeling of entitlement to free support.

In 2014, my wonderful son Noah was born and as surely many parents can confirm, it becomes somewhat harder to hack away into the night, so even though I've had plans to open source the Toolbox for years, I never managed to get this going.

Over the years, I also got contacted by literally dozens of people asking to help, but since I never reached a baseline, published open source repository we could have built upon together, I never could really give a pointer where to start.

### Why not simply open source the current version?

* The codebase is quite outdated since I didn't put much time into it beyond bare neccessities in recent years
* The app has for years also been my personal playground for trying out things, so generally it's just not in a shape that I consider good enough, neither for publishing with confidence nor as a good foundation for future development
* With the re-launch in 2011 I had added a whole lot of "community"-features like submitting resources about projects, comments, and more, most of which take up a huge portion of the current code yet I personally do not see these as part of a future Ruby Toolbox
* It also featured a way for the community to submit changes to projects using a Pull-Request-inspired mechanism. This was fully built into the app but as me as the only reviewer, also suffered from the same bottlenecks.

### Why is it so hard to build a new open source version?

It's definetely not hard to build a fresh version of the Ruby Toolbox, from a technical standpoint. 

More than anything though, every time I attempted to build a new version of the Toolbox, I stumbled into "philosophical" issues. Is this even needed? Who would use this nowadays?

### What should the Ruby Toolbox be?

Since I started the Toolbox in 2009, the Ruby ecosystem has changed, and stabilized, a lot. For many use cases, the community has settled on a specific solution. I would really like to hear what you all would like too see from the Ruby Toolbox. I won't go into much detail on my ideas in that regard here in order to not give too much direction into this discussion.

### Sustainability

I've experimented a bit with adding ads to the Toolbox in early years, but the 100 - 200 USD per month were in my opionion ultimately not worth polluting the site with banner ads. For a few years now, the site has had a Paypal donation button, which currently generates ~25 USD per month from 3 recurring donors and one-off donations between 0 and 15 USD per month. I've been taking care of paying the hosting, which has been around 60 EUR per month. 

I take great pride of providing a resource that has proven very useful to fellow developers all over the world, but it also is sometimes down-putting to, beyond time for development and maintenance, also add money to keep the site going.

After the outage, plenty of people suggested to create a more formalized donation scheme, for example with Patreon. Would people be interested in this? Ultimately, if we get enough donors I even could put in some proper work time in place of my usual freelance consulting gigs. I also wonder whether companies would be interested in joining in on this? Again, I'd love to hear your thoughts!

**I'm looking forward to your feedback, and thanks a lot for the many kind words of encouragement and support that reached me over the past week! Let's get this started!**

**Comment by a-b** [comment]
I like and use your project a lot. Talking to a community is a great starting point!

**Comment by punjab** [comment]
Thanks for all your hard work in maintaining RubyToolbox for all these years. It has been my go to app to check out library options.

**Comment by rubiii** [comment]
Thanks for opening this thread and for all your efforts to keep the toolbox running!

I think it’s a good idea to find corporations who would support a future version of the toolbox. Please leave a thumbs up if you agree and then maybe we can create a new issue to discuss this topic.

**Comment by jurgis** [comment]
You could also try out scaleway hosting https://www.scaleway.com/pricing/
It might bring down hosting expenses.

**Comment by PragTob** [comment]
:wave:

Thanks for opening up this discussion and for all the work over the years! :) :green_heart: :star2: :pray: 

Ruby toolbox has been great when it was working so I'd love to see a comeback.

Cost-wise, I'm not sure if this is applicable but mabe [ruby together](https://rubytogether.org/) would be interested in incorporating the project? I'd consider it part of the ruby ecosystem, I mean it's no ruby gems... but it's a nice accompanying guidebook :)

**Comment by robinboening** [comment]
I want to thank you as well. The RubyToolbox is a well known project and for me it was always a great place to check for alternative gems. I like PragTob's idea with ruby together - I think you should just talk to them.

**Comment by mediafinger** [comment]
After all these years I still keep checking the Ruby Toolbox to check which gem is the 'hottest shit' right now. And I have the impression others must do too, as I heard many people mentioning the site over time and in recent months. Some even wished to have a Toolbox like site for other languages.

As it is seen as such a great resource, I surely hope you/we will find a sustainable way forward. If that will be a complete contributor driven re-write without all time-consuming community features, a cheaper hosting solution, a Patreon financed rewrite or an organisation that would be happy to sponsor... who knows. When you decide to go the latter way, I would definitely contact all larger companies using Ruby that also sponsor conferences and ask directly for a 4 figure some per month for the next year for them to see their logo on the page.

I am happy that you are finally going into the direction of open source. I remember us talking about it many years ago. And you basically mentioning the same reasons that have hold your back from publishing the code base.

**Comment by waynerobinson** [comment]
Ruby Toolbox is a great resource, but I understand all your stated issues with running a free service like this. 

I'd be happy to pay a small subscription, but given the experience of Ruby Together who have the profile of the official org for both Ruby Gems and Bundler and yet to hit $20k/mth, I'm probably in the minority. 

Thankyou for this resource and I support whatever decision you decide to make going forward.

**Comment by SeriouslyAwesome** [comment]
I haven't used Ruby Toolbox in a few years, but it was instrumental when I was learning Ruby/Rails to figure out what gems were worth looking into. More recently in my career, I started learning Ember.js, which has [Ember Observer](https://emberobserver.com) which I immediately recognized as a modernized RubyToolbox for the Ember community. One of my favorite aspects of it is its review/evaluation system of each package, which asks the following questions (taken from their website): 

* Is the source accessible? - This one is straightforward: can the source be found? In practice this has turned out to not be a terribly useful question and will probably be removed at some point.

* Is it more than an empty addon? - Is the addon more than the base generated addon? Another question that turns out to not be very useful. It will probably be removed in favor of hiding addons from the site until they are more than an empty addon.

* Are there meaningful tests? - The key word here is "meaningful", in reviewing we took this to mean any tests beyond the tests provided by generators.

* Is the README filled out? - Does the README have content that is not the default generated content?

* Does the addon have a build? - Is there a continuous integration build? We answered N/A if there were not meaningful tests but there was a build.

This is probably more helpful to the still burgeoning ember community than the relatively stable Ruby community, but I dig the idea of identifying which gems have real effort going into them and which are just slapped together to get something done real quick.

**Comment by evaryont** [comment]
I've found Ruby Toolbox to be extremely useful as someone who hasn't paid too much attention to the ruby community for years and then coming back. For a variety of use cases, like you mentioned, the community has solidified on one particular choice. But how do I know which one that is? This is where I see Ruby Toolbox fitting in, for my use case. I'd love to see that particular functionality preserved in whatever incarnation comes from this.

**Comment by 3zcurdia** [comment]
First of all, I wanted to thank you for maintaining Ruby Toolbox I  appreciate the effort to keep it alive after all these years. Regarding the future, I think what we care most is the ability to compare statistics between different gems with the same or similar functionality. One of the options that I will like to see in the future is a distributable command line tool; I don't know how you collected the GitHub repo info, and the score but that could be done inside the distributable file.

**Comment by welkie** [comment]
I think a modern Ruby Toolbox still has a role and I'd be happy to make a small monthly donation towards upkeep and hosting.

Hearing that you were paying 60 EUR per month to host it alarms me. I've had an opportunity to learn more about technologies like AWS and Docker over the past year, and I do know of ways to get a web app hosted for much less, even with the app having multiple backing services. I'd be happy to share those techniques with you if you like.

**Comment by jamesmk** [comment]
The Ruby Toolbox was a great resource and I would find it useful today if it was still around. The core concept was solid, the only things I'd suggest would be an easier way to add new gems and maybe an API. 

If you started up an OS version I'd gladly contribute. Thanks for all the work you've put into it!

**Comment by phoet** [comment]
1. 💗
2. I disagree, you should just OSS it
3. I would be willing to contribute time, but not money
4. Have you talked to Rubytogether / Rubygether.org people about a joinedventure?

**Comment by PragTob** [comment]
:wave: btw. I reached out to Andre Arko from rubytogether and he said he'd bring it up in the next board meeting :)

**Comment by mediafinger** [comment]
Personally I don't belief that _rubytogether_ would be a viable option to get the Ruby Toolbox back quickly. AFAIK they bill 150 US$ per hour and don't do free work under their brand. As they get less than 20k per month, they just have around 120h/month to distribute _for all their projects_.  
I would assume the Ruby Toolbox needs a concerted effort to get back on production - if it should not be done with the current code base.

But let's see what they conclude in their own evaluation.

**Comment by kikito** [comment]
What I would do in your case is: release the source code as it is, find someone you can trust to deal with the community and maintain the project - Ideally someone who would benefit from the extra visibility but is still somewhat experienced. You assume the role of a benevolent dictator, only stepping in when there are conflicts / excessive bikeshedding / abuse (and maybe creating some basic issues for things that need updating to get the ball rolling).

To be clear, I am not volunteering to do the community stuff. I've got a 11-month son :)

**Comment by phoet** [comment]
@dwradcliffe do you think shopify could be interested in something like this? it sounds like a great hackdays project to me.

**Comment by sandstrom** [comment]
I think something simple with categories, scores (based on some metric) plus some Git/repo stats would go a long way. Simple search would also be useful.

I think these two websites are quite good (an addon is more or less the same thing as a gem in the Ember universe).

- https://emberobserver.com/
- https://www.emberaddons.com/

They are basically single-page apps that read from a Rails API backend (or similar).

(it may even be a good idea to fork one of them and use that as a basis for a new version of Ruby toolbox)

**Comment by broomie** [comment]
I have used The Toolbox for years and it was always helpful. I am happy to contribute to a fund if that helps

**Comment by mchilson** [comment]
**Thank you** for all your hard work over the years and I hope you will consider moving forward. RT has been a great help over the years helping me switch over to Ruby/Rails from other languages. Count me in to help out if you go open source. As you can see there is still a great amount of interest in the toolbox. Maybe do a kickstarted campaign so you can devote some hours to getting it back up on its feet 100%. Whatever you choose to do thanks again!

**Comment by colszowka** [comment]
Wow, thanks everyone for the great, encouraging responses so far! 

I'll try to spin off some of the topics brought up here into separate issues tomorrow so we can have more of a structured discussion on specific things there. Thanks again for your all your great input!

**Comment by bartiaco** [comment]
I just want to add in: I had noticed that Ruby Toolbox was languishing, but I've found always it an incredibly valuable resource. Especially useful when you're coming back to a category of gem and want to know what's the current state of the art.

Possibly even taking it beyond Ruby into JS and other languages would be great too.

**Comment by phoet** [comment]
@bartiaco is right, this is a crosscutting concern and I think that @andrew actually solved most of the problems with libraries.io

**Comment by andrew** [comment]
I've taken a lot of inspiration from Ruby toolbox with https://libraries.io, which tries to help solve open source discovery for every community, one area that I've not had chance to get up to speed on yet is categorisation. 

We're a non-profit project funded by the Sloan and Ford foundations and if there's anyway I can help do let me know, either by providing data to refill rubytoolbox or adding the same categories into libraries.io

**Comment by colszowka** [comment]
Thanks @andrew, your work on libraries.io is very impressive!

I think focusing on just this one programming language was actually providing a lot of value to users of the Toolbox. I guess it depends on the use case, if I just want an [awesome postgres CLI tool](https://github.com/dbcli/pgcli), then I don't care what it's built in. When I need authentication for my rails app, going with Python might be cumbersome. Even supporting this one language as a maintainer proved to be very difficuilt in the long run, it's just a lot of effort to maintain the categories in a sane way, keep up with new stuff etc, and even display stats that are most relevant in that exact domain.  Introducing and maintaining categorization (and a schema for moderating them) would be much harder for 30 package managers than just one programming language.

Two things the Toolbox (imho) did well in it's heyday was discoverability, through both the search and the categories, and usability - I always tried to show the most relevant stats front and center wherever you saw the project, so there was no need to click through to projects from overview lists.

Take for example the following case: I want to find a ruby code coverage library. On the Ruby Toolbox search, this showed up the most popular ruby gem simplecov first. When I search on libraries.io, I get:

* For "coverage", results for all programming languages - simplecov does not show up on the first page
* If I narrow down to Ruby as a language, Simplecov (3000+ stars) at 10th place - results 1 - 9 have less than 15 stars each (no further stats shown, I need to click through)
* For "ruby coverage", simplecov is 4th, projects shown first have all at most 16 stars, including a sublime plugin written in python
* What is also very confusing: If I click on Rubygems on the start page, search remains global. I'd expect it to be narrowed down to the language (maybe with that neat "deletable" prefix for "this repo only" like on github)

Don't get me wrong though, your site is awesome, the only thing I'm getting at is that I think there is a difference between a general-purpose open source database, collecting allthedata(tm) and a site that tries to help aid in finding what's used by everyone else for any given problem.

Anyway, the data that you are syncing is definetely also what the Toolbox is using. @andrew Could you maybe give more details on the "Open data release coming soon." that is announced on the front page? It might indeed make a lot of sense to use your db as the backbone? On the other hand, I'm a bit skeptical about using a "proxy" API instead of the proper upstream APIs, since there would be an additional moving part that could break, and keeping in sync might be tough (consider rubygem release webhooks for example), but that all depends on what integrations you have planned :)

**Comment by andrew** [comment]
Our open data release is now public, blogpost here: https://medium.com/@BenJam/libraries-io-releases-data-on-over-25m-software-repositories-ab1db665826e, more info and documentation here: https://libraries.io/data and the actual csv downloads are here: https://zenodo.org/record/808273

We also have an API on the site here: https://libraries.io/api, if there's anything missing we can definitely add it, we've also got a basic streaming API where you can listen for new versions, which will be beefed up with all repository events over the coming months.

All the data is available under a CC-BY-SA license and all the code for Libraries.io is open source too.

Agreed, our search isn't great at the moment, we've been focusing on the goals of our grants which mostly involve collecting more and better quality data (full list of grant goals here: https://github.com/librariesio/supporters/issues), but discovery is very important to me, it's why I first launched the site.

Another area that I see happening a lot with open source is a developer builds a tool for one community (usually centered around a package manager) but it's too closely tied to the implementation and can't be reused by any other communities, so there ends up being a lot of rework and a lot of dead projects. 

I've been trying to normalize the data for all the different package managers and source code repositories to help people build tools on top of the data that work for everyone, as well as avoiding a GitHub based mono-culture, which is why I've spent a lot of time adding support for collecting metrics from GitLab and Bitbucket repositories as well.

When it comes to the categorization, it's potentially quite a fun problem to solve to try and get communities to categorize new and updated projects as they are released, potentially with some automated classification based on keywords, etc

Happy to work with you in whichever way you think is best!

**Comment by colszowka** [comment]
@andrew That sounds awesome, I know by heart how hard aggregating this data is, really great work! This might be indeed extremely helpful, in whatever way we'll proceed. I'll get back to you, but thanks a lot already for your input!

**Comment by andrew** [comment]
ccing @sclinede here too as he's been working in a similar area of ruby tooling assessment: https://ossert.evilmartians.io/

**Comment by kddeisz** [comment]
Maybe talk to Heroku? I wonder if they could give you a deal on hosting costs.

**Comment by sarahmei** [comment]
Thanks for all your work on the toolbox over the years. It's been such a great resource. I talked about it in my Making Technical Decisions talk (now several years old) as a way to highlight the social data people consider when they're thinking about which library to use.

I'm a big fan of Andrew's work on libraries.io, and the funding model there is also super interesting. I'm happy to see that discussion already started. :) I'm also on the board of Ruby together, and would be happy to chat with you before our meeting to get a sense of where you are & what would be helpful.

**Comment by vassilevsky** [comment]
It seems that the Ruby community will be better off with discoverability / rating features integrated into rubygems.org instead. It is **the** website will all the gems, so why would someone want a separate site with additional information about them? RubyGems is aleady open source and has great infrastructure, metrics, statistics, etc.

**Comment by thejamespinto** [comment]
@colszowka I'm one of the people who have buzzed you about outsourcing RTB over the years.
Whatever you choose to do, man...
Look at how many interactions and responses you had in 48 hours.
Whatever you choose to do, you're gonna have people interested in helping you.

Also, thanks for putting up some version of the database up on the current site.
It means a lot to me because I needed to check some categories.

We love you ❤️

**Comment by welkie** [comment]
@vassilevsky I think that it's better to have them as separate sites.

rubygems.org has a narrow focus and it does what it does well. It acts as a repository for gems and includes data about the gems like links to their main website, etc. It doesn't tackle social aspects of gems and subjective concerns (like "What's the best ORM for Postgres?") because to keep up with those concerns would be a monstrous task.

This frees up the community to put together sites that cater to the social aspect of gems including subjective concerns. And if one site doesn't do well, another is free to crop up and take its place.

**Comment by markets** [comment]
Hello everyone,

First of all, thank you a lot @colszowka for your awesome contributions to the Ruby ecosystem 👍  👍 👍 

I'm maintaining the Awesome Ruby project: https://github.com/markets/awesome-ruby. The nature of both projects are quite different, in Awesome Ruby there is practically no quantitative numbers and all additions/deletions are managed via GH pull-requests, comments and reactions. This is cool, it's pretty easy for the community to contribute and discuss (to be honest would be nice to see more movements there) and this helps a lot to maintain the collection.

So I think both projects can co-exist. Would be very amazing to see The Ruby Toolbox up again with some updated stats. I'm sure a lot of people would be happy to help maintaining an open source version.

There is also https://ruby.libhunt.com, based on the Awesome Ruby collection and built as a web app with stats, comparisons, newsletters... but isn't open source.

The Ruby Toolbox it was a 🔝  reference for years, thanks!

**Comment by MadBomber** [comment]
I wet to the toolbox within the last two weeks looking for recommendations on some to support accessing an old SOAP interface.  It has been a good resource.  I do not want to see it go away.  I want to be able to go to it and if I do not see an existing category with recommendtaions I want to be able to ask the community "got anything that will do xyzzy?"

**Comment by rckv** [comment]
@colszowka Thank you a lot!
Personally, I favor the idea of integrating with [RubyGems.org](https://rubygems.org) by adding statics, classification, recommendation, etc.

**Comment by phrozen77** [comment]
Keep the site up, please :)

While i'm not a Ruby dev for long (if i can even call myself one at this point ;) ), in my previous attempts at becoming one, The Ruby Toolbox was a great resource to find what i need, gem-wise...

I'll keep an eye on this and will help in whichever way possible :)

**Comment by leonardk1** [comment]
@colszowka , thank you for your work on the rubytoolbox. it has been an invaluable tool for me.

I can contribute by actually providing time to do the actual coding :)

**Comment by arbox** [comment]
I second @rckv and would help with integrating RubyToolbox into the Rubygems.

**Comment by xliiauo** [comment]
Well, as a professional ruby developer since start of my career, the toolbox is my goto place when I need something. It's like a real toolbox, and I use it a lot. Thank you for your contribution for the years. 😄

**Comment by colszowka** [comment]
Hello again, and again thanks for the amazing response :)

Generally, it seems people did indeed find the Ruby Toolbox useful :) So, as far as I can see we have three options:

## 1. Integrate the discovery features into Rubygems.org

As [suggested](https://github.com/rubytoolbox/rubytoolbox/issues/1#issuecomment-308819946) by @vassilevsky and seconded by a few others.

This would require that rubygems.org pulls the relevant repo data from github (and potentially other source code hosts) periodically.

First of all I have no idea whether the Rubygems.org team is interested in this kind of thing. Not sure who to ping here, let's try @dwradcliffe and @indirect. @sonalkr132 seems to have been pretty active on [improving the Rubygems.org search recently](https://hackernoon.com/you-know-for-search-de62b6126a86), which is somewhat related.

Personally, I second what @welkie [said in that regard](https://github.com/rubytoolbox/rubytoolbox/issues/1#issuecomment-308847414), in that rubygems should be focused on, you know, hosting packages. It's surely a tough problem to solve on its own, and personally I think it's not a package manager websites role to "play favorites", it should be a neutral place that just gives you the damn packages.

## 2. Close down and hand over to another service

Notable comments here are:

* @markets [regarding awesome-ruby](https://github.com/rubytoolbox/rubytoolbox/issues/1#issuecomment-308850898) and the (closed-source ) [libhunt that is based off it, but not directly affiliated](https://ruby.libhunt.com/)
* @andrew [regarding libraries.io](https://github.com/rubytoolbox/rubytoolbox/issues/1#issuecomment-308729759)
* @andrew also mentioned https://ossert.evilmartians.io/ by @sclinede, which has very impressive mechanisms for rating libraries on maintenance and other aspects. It's really nice, check it out!

Basically, Ruby Toolbox closes down and all people interested move over to another service that provides similar utilities for discovering stuff, and try to build in missing features (like categorization and better search on libraries.io, again see [comment by](https://github.com/rubytoolbox/rubytoolbox/issues/1#issuecomment-308729759) @andrew )

As @markets stated, the plain **awesome lists** are lacking the searchability / dynamic metrics due to their static nature. Of course, these lists can be further used by downstream service like libhunt.com, which is apparently run by a STANSOFT Inc., I could not find a github contact on the site. 

This approach only is capable of finding gems already categorized. I've personally used the Ruby Toolbox's search to find the best option for stuff that does not directly fit a category (i.e. password hashing comes to mind), and for that global gem search is needed.

**Libraries.io** is building an enormous, cross-language open database of open source software. Adding categorization and improving search is on the roadmap as @andrew stated. Due to the huge size of the platform, [I think it'll be much harder](https://github.com/rubytoolbox/rubytoolbox/issues/1#issuecomment-308662067) and time-consuming to build something with "optimal" UX for ruby specifically. I believe a narrower focus can help in the overall experience, as I stated above. On the other hand, [again linking back to @andrew](https://github.com/andrew), it definetely is a very noble goal to solve this once and for all. 

## 3. Re-build a new, truly open Ruby Toolbox

I won't go much into detail here about what that might specifically look like, generally many people have voiced a desire for a focused, straight-forward rebuild of the Toolbox, with categories, search and generally a quick way to compare and explore libraries.

Under the assumption (a.k.a. personal opinion) (see 1.) that a separation of concerns between the actual package manager and rankings/"social stuff" makes sense, and (again, personal opinion) that getting the optimal experience for libraries.io might end up being an enormous task (see 2.) this seems like the best idea to me. 

Here, I'd consider libraries.io the "metadata" library, and we could build on their excellent collection of data to build a ruby-specific solution, which (see 2.) seems like a much more conquerable problem. Basing off of libraries.io would have the additional benefit that other language communities could base off their own variants off our code base, so we ultimately would have something that would benefit everyone else, too.

---

This got long. so let's sum things up: Personally, as I wrote, 3. sounds the most reasonable to me, but hey, I'm the one whose beloved baby is at stake here, so you probably shouldn't be listening to me anyway... So, looking forward to your thoughs!

As a side note, over the past week a handful of development teams / companies have reached out, proposing to take over the Ruby Toolbox and re-build it. Although I certainly appreciate the offers, I am not interested in this. If we re-build, I actually and definetely do want to play a role in this whole thing, I loved doing this, and getting this kind of feedback is just brilliant! And also, if we re-build, I want this to finally be a fully open, community-driven effort.

Also, regarding the "sustainability" topic: 

* A few people here have said they'd be willing to give a monthly donation
* So far, I haven't heard from any companies
* @PragTob got in touch with @indirect regarding Ruby Together and @sarahmei from Ruby Together also [was seen here](https://github.com/rubytoolbox/rubytoolbox/issues/1#issuecomment-308778371) - I will send you three an e-mail (I hope I can figure out all your mail addresses ;)

Finally, thanks a lot @triskweline for offering to take care of hosting! I will get back to you on this when the time comes ;)

That's it for now, take care :heart:

**Comment by sarahmei** [comment]
@colszowka my github user at gmail :)

**Comment by sysadm** [comment]
Probably most convinience way will be reopen ruby-toolbox site from the last available backup and revert any possible content with wayback machine, for example: https://web.archive.org/web/*/https://www.ruby-toolbox.com/

And, of course in parallel, design new version of ruby toolbox. I'm absolutely sure that proposed variant 3. Re-build a new, truly open Ruby Toolbox - is the best. It takes some time I assume, so shutting down ruby-toolbox through all that time can destroy popularity. We can help you with hosting, backups, even development of new version.

**Comment by galori** [comment]
There is 

# 4. Open Source what you currently have and build out from there:

You be the arbiter of which PR's get merged and which don't but trust the community to help clean it up. Look past whatever state of mess it currently is in (I bet it's not as bad as you think it is, and even if it is that's ok)

A good mode for this is Textmate and Textmate 2. Textmate 2 languished without any updates for years despite having a huge fan base of people who just loved it and kept waiting.

When the developer of Textmate finally open sourced it, a huge community of people stepped in and helped modernize it and clean up the source code at breakneck speed. It's been in very-active-dev-mode since then. And I went back to using it - Textmate 2 is awesome.

An important thing to note with the Textmate comparison is that all this happened while _maintaining the Textmate license cost_. You still have to pay to buy the license to use it (one time cost). The community was ok with it - in fact that's what they wanted so that the original developer doesn't lose that income stream since it was his creation. My point is the open sourcing it is not incompatible with having a patreon or other donation/ supporter revenue. In fact in this case it is even more justifyable because you are still responsible for running the site.

**Comment by dwradcliffe** [comment]
First I'd like to say thank you because like many others I used this site quite a lot back in the day. :)

As for moving the functionality into RubyGems.org itself:
I agree with @welkie too. I would like to keep this separate. We've been able to keep RubyGems.org reliable and stable for a while now because we're very picky about what we add to the site. We have a better search system now, and I'm on board to improve search even more, but only with the data we already have. I don't want to be scraping in data from other places and I definitely don't want to be curating categories and picking favorites for gems.

**Comment by tkz79** [comment]
Thanks for all your work, it was so helpful getting me going... and I still prefer it when I want to find some new gems to use / learn from!

I would be happy to contribute time, open source or not... and even a little money. Have you thought about small annual donations like $6 or $12 / yr? I threw some at railscasts not too long ago to thank Ryan for his contribution to my journey, you definitely deserve some too!

I agree that banner ads suck... though not all advertising has to... in a community like this, I am pretty sure tucked away advertising could work quite well...

That said, if you ran a single banner with your own revive server (or something else non spying)... reserved for dedicated sponsorships / promos, I would gladly turn off adblock for you - I actually think I would look forward to seeing them. 

Mobile has destroyed online advertising for most, but i imagine your desktop percentage is amazing relatively speaking... if you keep it discrete, i think you've got a great occasional revenue source.

Some tucked away ideas:

Ex 1: Some resource pages for stuff like great hosts and services using affiliate links... heck... even a plain amazon link to shop through...  

Ex 2: A list of all the great ruby books and other resources out there with affiliate links where possible... 

Ex 3: A paid job board... something cheap like $10 or $20 / posting.. Can keep it dead simple to see if there is demand for it... and build it out if there is.  Users would likely subscribe to get emails for those job posts, might even donate!

Ex 4: A guide section to bring exposure to the resources... like how to setup your own VPS to avoid going broke hosting on heroku... great  product placement.... a welcomed ad... and great clickbait :D

Ex 5: A few donator benefits... perhaps the ability to tag gems that you'd like an email notification on updates...  the ability to comment (should cut spam right down! - 2 birds...)... perhaps premium access to more advanced guides... 

Would be happy to discuss ideas further.

**Comment by galori** [comment]
> Ex 3: A paid job board... something cheap like $10 or $20 / posting.. Can keep it dead simple to see if there is demand for it... and build it out if there is. Users would likely subscribe to get emails for those job posts, might even donate!

Actually this one is a really good idea. Companies that are hiring ruby devs have a really hard time...reaching all the ruby devs. Having a place for job postings that is so targeted and focused would be really valuable. I'm sure my employer would post there.

**Comment by sysadm** [comment]
With all respect, @galori, $400/post is too much, and probably you saying so, 'cause you are not employer.

**Comment by tkz79** [comment]
Cheap makes it easy to get some initial traction... especially if the initial goal is to cover hosting costs and hopefully some beer (vodka?) money. $400 could be the sale price one day... but I'd hold off on cranking it until both quantity and quality of leads can be measured.:)

**Comment by sandstrom** [comment]
What is a realistic number for the hosting costs? With the right architecture (single-page app, html/css hosted on S3, CloudFront or Fastly) it shouldn't be that much, right?

There is quite a lot of work to run a job-board. May be easier to just fund via sponsoring? Company sponsoring with logo + link in the footer. I'm thinking that a couple of hundred a month should be enough.

**Comment by colszowka** [comment]
@andrew I just noticed you do not seem to be tracking gem download counts on libraries.io, are there any plans on doing so? This is a pretty vital part of the stats the Toolbox uses for the ranking

**Comment by andrew** [comment]
@colszowka not yet but it's on the roadmap: https://github.com/librariesio/libraries.io/issues/53 can make it higher priority if you need it.

**Comment by NicolasWebDev** [comment]
Hi @colszowka.

Thanks a lot for your awesome work throughout the years! I was also one of the people contacting you on twitter about open-sourcing ^^.

I really think that The Ruby Toolbox is a cornerstone of the ruby community, and is a required tool for everybody beginning with Ruby, so I am pretty happy to learn about your decision.

As for my opinion on the matter, I would think that a rewrite would be the best bet on the long run. Maybe the data itself could be pulled from libraries.io, and the features included in a progressive way.
About the paid job board, I think it would be best to keep things focused to begin with.

Surely using AWS, paying only for the actual needed resources would lower the cost (as suggested above). Also I am of the opinion that with the involvement that Heroku has shown to ruby over the years, they would not mind sponsoring the project, for a mere $60/month.

**Comment by buzzware** [comment]
+1 for the primary place this site holds in my view of the Ruby ecosystem

**Comment by bradland** [comment]
@colszowka First of all, thank you so much for maintaining Ruby Toolbox on your own all these years. It has been a tremendous asset to the community. It is easy to overlook the "social" aspect of open source software, and Ruby Toolbox did a commendable job of providing discoverability and insight in to quality/popularity.

As you contemplate your choices, I would ask that you consider the timeliness of the solutions. Every day that passes is a day the community lacks access to this valuable resource. To compound that, participants in the Ruby Toolbox (anyone who suggested, rated, or otherwise contributed) have now lost access to information that they assisted (in small part) in creating. The more time that goes by, the greater the sense of loss will become.

Also consider the timeliness of the data you have collected in Ruby Toolbox. Gems change all the time, and Ruby Toolbox was a popular resource for curating the ebb and flow of Gems, especially in the more popular categories.

Thanks again for everything!

**Comment by spickermann** [comment]
I agree with @bradland, timeliness is important. IMO the best option is to restore the latest version and modernize the current app with the help of the community. A rewrite would take to much time and the Ruby Toolbox is too important to not be available for a longer period of time.

Count me in if you need help with the current application or a new Open Source version of the Ruby Toolbox.

**Comment by oddlyfunctional** [comment]
+1

**Comment by thejonanshow** [comment]
@colszowka I'm a Developer Advocate at Heroku and I have been meaning to reach out to you for some time to talk about the possibility of providing hosting for Ruby Toolbox. Perhaps this is the perfect opportunity.

If the community is comfortable with the idea I am willing to take responsibility for getting the code, as it is today, up and running on Heroku, including any costs associated with that.

Ideally this arrangement includes a small "sponsored by Heroku" badge on the homepage, but again I will yield to the community. My priority there is to make sure this is an easy sell to future Herokai long after I'm gone.

If you have any questions that you'd prefer not to discuss publicly please feel free to email me, I'm jonan@heroku. The same goes for anyone else who might have feedback on that plan.

At the very least the community will again have access to this invaluable resource while we band together and work on the grand rewrite.

**Comment by dixpac** [comment]
I think that open sourcing the code that you already have + generous offer from Heroku and @thejamespinto is the best path at the moment.

Then you could create team of the maintainers, draw roadmap, add issues to Github and let the community with the maintainers take over :)

I think this project is valuable to the ruby community and shouldn't die. I would be willing to help also :)

**Comment by welkie** [comment]
+1 to open sourcing current code and accepting Heroku's offer! :)

**Comment by replicat0r** [comment]
+1 to open source the code and accept the offer. Ruby Toolbox has been an incredible tool over the years. Its the first place i usually check. Would love to contribute and support

**Comment by colszowka** [comment]
@thejonanshow Thanks a lot, I'll get in touch via mail soon!

In general, I agree with what many said here that we should get an interim solution as this discussion goes on and hopefully evolves into something new.

**Comment by t0nylombardi** [comment]
Open up a slack and let's write up a wireframe and start writing logic to that wireframe.

**Comment by colszowka** [comment]
@t0nylombardi I prefer the async nature of github issues for general planning, seems much more effort to follow unthreaded, real-time discussions on slack instead of just catching up with github issues on specific topics

**Comment by varyonic** [comment]
I used to love Ruby-Toolbox and appreciate the effort that went into it.  It was a great project, truly useful.  Looking at [this thread](http://help.ruby-toolbox.com/discussions/problems/1790-is-ruby-toolbox-dead-it-doesnt-seem-to-be-actively-maintained-any-more), however, there were concerns about ruby-toolbox abandonment in September 2014.  I was subscribed to this topic awaiting updates and involvement for years.  There was an upbeat update in March 2015, and then nothing... for two years.  I've moved on to using [Awesome Ruby](https://ruby.libhunt.com/), [OSSERT](https://ossert.evilmartians.io/) and [libraries.io](https://libraries.io/rubygems), all great projects.  A project that is maintained is better than one where the maintainer is too proud to talk to or hand-off to those with capacity to help.  I'm here to say reconsider option 2: the ruby-toolbox name has lost its reputation and it's time to move on.

**Comment by anonoz** [comment]
Wait, there are other sites other than ruby-toolbox.com? I've been using this and exclusively this for a while.

C'mon let's do it for the new coming rubyists!

**Comment by rainchen** [comment]
ruby-toolbox.com is very helpful to me. It's really useful if Rubygems can "merge" ruby-toolbox, it's missing a "Category" entrance.

**Comment by padi** [comment]
While the alternatives are useful, Ruby Toolbox is still my go-to place for categorization of gems.

Thank you for your hardwork, @colszowka. Personally, I prefer option 4 and/or taking up @thejonanshow's offer, but I'd love to see it continue, whichever path you choose.

**Comment by revans** [comment]
@colszowka I'd be happy to help with the writing of a new version. Hit me up, if this is something you want/need: robert@codewranglers.org.

**Comment by tiredpixel** [comment]
@colszowka, first, thank you for your efforts over the years. Regardless of the path taken in the future, I for one have found this a useful resource numerous times in years gone past. (Indeed, I just found this message when I was attempting to use it to search for a library I'd forgotten the name of.)

Could you tell me more about your hosting requirements, please? What did the `$60/month` give you? What did the stack look like?

**Comment by toobulkeh** [comment]
Sounds like a quickfix would be to go with Heroku, while a longer-term solution would be to merge functionality with rubygems / Ruby Together.

This original post implies discussing a longer-term solution, but the action / direct communication with Heroku implies the quickfix.

👍  for a longer-term solution.

**Comment by bradland** [comment]
@toobulkeh this won't be merged with Rubygems. Excerpt:

> **dwradcliffe:** I would like to keep this separate. We've been able to keep RubyGems.org reliable and stable for a while now because we're very picky about what we add to the site.

See [this comment for the full post](https://github.com/rubytoolbox/rubytoolbox/issues/1#issuecomment-309128895).

**Comment by tkz79** [comment]
I'm thinking it might be a good time to break this issue up... I personally would like to stay in the loop, but getting notified every time someone sends a thanks along is getting to be a bit much.

#1) Closed with links to new issues along the lines of:

#2) Thanks/Praise to @colszowka and anyone else who contributed (just to keep the other issues clean of clutter)

#3) Stack / Hosting

#4) Core Functionality

#5) Secondary Functionality (Monetization)

#6) List Ongoing Tasks (server ops, curation, moderation, etc...)

#7) Volunteer / Contributor List

**Comment by chupa-cabras** [comment]
Saudades do Ruby Toolbox

**Comment by shevegen** [comment]
> I would really like to hear what you all would like too see from the Ruby Toolbox.

I think that the overview from ruby toolbox was very useful and it is probably still better than rubygems.org

Personally though, and this is from a usability point of view but also philosophical point of view, I think it would be better if it would be integrated into rubygems.org itself. Whether it is a separate code or integrated is secondary there - I mean in regards to users.

I guess the rubygems.org maintainers do not want to add more code and maintain that. I can understand it from all point of views really.

From a user point of view, having a summary-page that can compare different projects like ruby toolbox did, can be useful. What may help is to also get more feedback from users that way - I use rubygems.org a lot but I rarely use github myself in the sense of issue->bug fixing, for somewhat similar reasons as colszowka - I would not have the time. (I am also not a professional developer so all the time I put into ruby is really more a hobby or things that I need myself; and sometimes that may be useful for other people too, under a permissive license.)

So for my own personal use cases, rubygems.org works. And the search feature/filter has recently been improved, which is also good.

I guess others can not estimate how much time and effort you may want to invest, and it may be that the donations will probably not hugely increase. I don't speak of experience here myself though - HOWEVER had, I also know how much money twitch game streamers make, even for games as old as warcraft 3... it's pretty insane in some ways. Would be cool if one could work on the code, teach people at the same time AND make money through open source. There are definitely projects where this or similar strategies work (look at all the kickstarter-games ... that's quite insane really) but I am not sure if ruby toolbox would be a prime candidate to qualify for it. But that's just my 2 cc anyway, feel free to disregard it. :)

TL;DR: From a user's point of view, it would be nice to have some integrated search-result-comparison functionality that ruby toolbox could offer.

**Comment by coisnepe** [comment]
So is @colszowka awol again?

**Comment by danielvu** [comment]
@colszowka Engine Yard has a focus on getting back involved with the Ruby on Rails community and this seems to be a great opportunity. WE would love to help here so can we connect to discuss the needs here a bit more?

**Comment by arbox** [comment]
I still think that a least the categorization part should be done inside Rubygems. The Rubygems specification should get an additional field for categorization metadata.
All projects like Libhunt or RubyToolbox need that! Why not have a central source provided by the authors of particular gems who know their work best?
Then we can enrich, cluster, combine or whatever we need, calculate popularity metrics...
Categorization is simply there in every package manager from deb/rpm up to npm packages. Even GitHub asks to add tags to repositories.

**Comment by colszowka** [comment]
@arbox The problem about getting metadata from Rubygems (especially so when introducing a whole new  metadata field) is getting gem authors and maintainers to actually add and maintain the field, keeping a high data consistency (what if someone makes a typo in "Web App Farmeworks"?)

I agree that this would be wonderful, and it should be technically possible with the newly added metadata fields, but we would at least need an interim solution that relies on manual / 3rd-party categorization but allows Gem owners to define their categories along the way.

**Comment by colszowka** [comment]
Also, I was off sick for the past week, but slowly getting better.

I agree with @tkz79 that we should probably wind this thread down and create more specific followup issues, I'll try to get this done over this weekend 🤞

**Comment by arbox** [comment]
@colszowka trust me as an nlp guy it's a much smaller problem to normalize wording between Farmeworks vs. Frameworks than to be a wizard and to collect non existent keywords :smiley: 

And it would be wonderful if we could do the following two steps:
- add a metadata field and a warning about missing keywords;
- add in the near future a blocking obligation to add keywords during the build.

And then automatic/semi-automatic 3rd-party services (or Rubygems itself in the index) could add  categorization on the top of keywords/other textual data/Gemfile clusters/Download correlations and so on.

If we won't crowd source the categorization and delegate the first step to the Gem maintainers themselves no automatic solution will be sufficient.

**Comment by Wokoro** [comment]
Thank you so much for your time, resource and energy you have involved into this valuable online resource, please this resource is too valuable to developers both in the beginner, intermediary and advance levels. 

please all efforts required to sustain this resource should be taken.

**Comment by franklinyu** [comment]
@arbox quote from @colszowka

> …we would at least need an interim solution that relies on manual / 3rd-party categorization but allows Gem owners to define their categories along the way.

And I agree. “Automatic solution” is a long-run solution, but we need something *now*.

**Comment by colszowka** [comment]
Hey everyone, as promised I got started on spawning some followup issues from this one. I think we have more topics to be discussed, but I'm still slightly ill and running out of weekend. Two topics that still need their own place are: "Hosting" and "Sponsorships / Sustainability". I'm sure there are more, but I don't want to start empty placeholders for the sake of having opened an issue, I think all of the topics deserve at least a broad introduction.

In general, #2 "Bring back old site" should be pretty self-explanatory, while #3, #4 and #5 can be considered as an invitation to discuss and describe what you would new or existing you'd like to see on a new ruby toolbox in the future.

Thanks :)

**Comment by mpz** [comment]
thank you for toolbox

**Comment by kronn** [comment]
So this is now the the official "thank you"-issue?

Thank you for the ruby-toolbox. It has been a valuable resource, especially in judging the "activeness" of projects. Or just to discover "what's the current solution(-landscape) for \@common_problem".

**Comment by emilebosch** [comment]
@colszowka I've been contacting you some time ago to rebrand and rebuild  ruby toolbox. I also offered up my designer to maybe rebrand the designs if you want to. I remember back then that you wanted to relaunch, is this maybe still a good idea to get this thing going?

**Comment by rainchen** [comment]
@emilebosch I would like to contribute coding if this becomes a open source project, FYI

**Comment by imnithin** [comment]
This could be my first open-source contribution. I'm interested to contribute if it's an open-source.

**Comment by matma6** [comment]
From my POV this page is like rubygems, but with a lot of order. So, e.g. I wanna find a web framework. On your page is one click. On rubygems I can type "web framework", but result is piece of crap. When I have to write some ruby code I need your page, because I need gem recomendations!

**Comment by lethunder** [comment]
I agree with matma6

**Comment by joshuapinter** [comment]
> Is this even needed? Who would use this nowadays?

Umm, myself and every other Ruby and Rails developer that I know. Trust me when I say that this is the best resource for discovering Ruby libraries to solve problems without re-inventing the wheel. The amount of time that I have personally saved because Ruby Toolbox exists is nearly uncountable.

This same approach has been replicated for every other language and framework I've moved into, including React and React Native. (The most recent being [Native Directory](http://native.directory))

**This is an important project and resource and should absolutely be continued in some fashion.** Would love to help out where I can as well. 

Thanks for all your hard work and sorry to hear about the server crashing.

**Comment by adesakin** [comment]
What is the main need right now? Funding to host the site or funding to build?

**Comment by tiredpixel** [comment]
Indeed. And per my previous question, what are the actual hosting requirements?

**Comment by etewiah** [comment]
I've posted on hackernews about this:

https://news.ycombinator.com/item?id=14831459

And on reddit:
https://www.reddit.com/r/rails/comments/6p0g4b/rubytoolbox_great_resource_for_rails_at_risk_of/

I think it will help to make as many people aware of this as possible  and build some momentum for a funding campaign.

**Comment by robbanp** [comment]
I might be able to help. What do you need?

**Comment by chrishough** [comment]
Happy to help where I can. I used this site all the time for comparing and contrasting gems. So far the only solution I have tried is http://bestgems.org/

**Comment by umerfarooq** [comment]
@colszowka I have always wanted to Thank you for putting the Toolbox together and maintaining it over the years. It was always my go-to site to do the comparisons and look for the alternatives when I was learning Ruby (& Rails) back in 2008 (never liked using  rubygems.org back then) and still ask and refer my employees/students to use it. I'd totally love to see it running back in full strength and would always be available to help in whatever capacity I could possibly do so.

Love and regards from Pakistan's Rubyists Community

**Comment by xecutioner** [comment]
Thank you for all the years in helping . OSS it for sure. I am sure someone might pick up .

**Comment by ufo2mstar** [comment]
Really appreciate your effort on this project @colszowka !
I benefitted a lot from this site (more than I can count), and I felt the info here gave a good high-level overview of all relevant tools that are popular with the community for any given task, than disparate sources and people's opinions.
As you feel that the technical limitations of a redesign/reimplementation are minimal, it would be great to see this project open sourced because I feel that like me, so many others will be in a position where they can contribute their own time and effort more than money on a regular (or one-time) basis.

**Comment by chrishough** [comment]
@colszowka have we considered https://www.patreon.com/?

**Comment by decal** [comment]
I can offer up one or more VPSen for hosting, since both members of the family that were hosting their business on them have unfortunately passed away.  To give you an idea of the boxes I have at my disposal, that are currently going unutilized, one of them is an 8-core CPU, 6GB RAM, 100GB Disk with CentOS Linux release 7.3.1611 (Core).  There are others running FreeBSD and Windows Server 2016 as well.  I'd hate to see such a useful resource to the Ruby development community be forced off the Internet for such silly reasons.. I'll give you the access--all you have to do is maintain the site and behave.  If interested, feel free to contact me via e-mail at: [[decal (AT) programmer {DOT}
 net](mailto:decal@programmer.net "decal (AT) programmer {DOT} net")]

**Comment by tvanderpol** [comment]
I don't have much to add to the discussion, though I'll follow it with interest and would consider time or money in support. I just wanted to say ruby toolbox remains basically my first stop when I dig into a new area of ruby or rails gems - it might not always by itself be sufficient info these days but it's still by far the best place to get a quick lay of the land.

**Comment by emilebosch** [comment]
Hi all, so many people want this to be a thing. 🙌  Can we please make steps? If not i'd consider closing this issue. Sorry for being so rash _sorry not sorry_, but i think it is proven that people want/need something like this. So building it is paramount, and i don't think we really need to talk a lot more about it, we need to build it.

I was thinking of maybe organizing a big hackathon, and then just hack it over the course of a weekend. Or at least launch something? I don't mind helping organizing it. I'm not saying it will be done then but a start is made.

If the hosting costs of 60 a month is an issue. I can chip in 30 and i'm pretty sure my founders or my fellow devs are also willing to pay. So i don't think thats the issue, lets just build the thing and stop bikeshedding. Thanks a lot <3

**Comment by chrishough** [comment]
I do not think building `new` is the issue, rather the old statistically data that made the toolbox what it was.  A clone without it, would not be very valuable at all.

**Comment by emilebosch** [comment]
Why not? I believe the aggregators and counters werent up to date and a lot of it was stale anyway. Not launching and building it because the initial data is missing seems like a missed opportunity. I dont think its impossible to build it up again.

Nowadays with github dependents and gem info we can sort of rebuild it again clean slate

**Comment by colszowka** [comment]
> Nowadays with github dependents and gem info we can sort of rebuild it 
> again clean slate

And that is what we'll do. Real soon now.

**Comment by deepakmahakale** [comment]
Access the ruby-toolbox site from 16th May 2017 here 

https://web.archive.org/web/20170430093635/https://www.ruby-toolbox.com/

**Comment by oasic** [comment]
@colszowka  Thanks for creating this site that the Ruby community (myself included) has appreciated for years!  Based off the publicly available traffic stats for Ruby Toolbox, you should be able to get $250 - $600/mo using AdSense alone.  I suspect there would be other opportunities for additional revenue as well given your position in reaching out to developers.  A couple banner ads on a page is not going to degrade the site.  In fact, I see it as the opposite -- the revenue justifies and incentivizes the time necessary to make the site even better.

**Comment by jasonfb** [comment]
@colszowka -- We use it! We need it! We love it! You're awesome!

**Comment by ekampp** [comment]
I love the site. I would love to be part of updating the code and even donating. I wasn't aware that the costs of running it was that high.

I don't think that a tool like this will benefit from adds, so I will happily be a Patreon for a few bucks a month. And I suspect -- based on the level of love in this thread -- that others will to. 

Perhaps if we start an open source version, we can also work on getting the price down by selecting better an cheeper tools as a community?

**Comment by andyjeffries** [comment]
Out of interest, how much server resources do you need? In terms of number of CPU cores, RAM, disk space? I'm the CTO of the company behind https://www.civo.com and while we haven't talked internally about sponsoring OSS projects with free hosting, I'd be happy to have a conversation with you first about whether it's feasible... Feel free to drop me a mail - andy@absolutedevops.io

**Comment by gobijan** [comment]
Just found this meta-ripoff of the toolbox that works for many languages.
https://ruby.libhunt.com/

**Comment by weedySeaDragon** [comment]
One of the **_really valuable_** things that Ruby Toolbox had was the at-a-glance information about
   - when was the gem last updated
   - how actively it is being maintained / worked on
   - number of uses/installs

That is what constantly helped me wade through lists of gems that all claimed they could be helpful.  

**_That_** is what distinguished Ruby-Toolbox from other lists of gems ('awesome' or otherwise). 

My input is that this is a key feature that must be included.

Perhaps this would be something that we could start on now?  Some folks could put together some different UI possibilities, and some can work on the 'getting the data' part.  Maybe folks can start with the existing code, or maybe there's value is also starting from scratch..... ?

**Comment by colszowka** [comment]
Hello again everyone,

time for an update! As you might have already [heard][tweet-1] on [twitter][tweet-2] last thursday, @rubytogether have offered to sponsor my work on re-building the Toolbox and will be funding around 5 hours of my work on that per week - awesome! I was in Budapest for EuRuKo over the weekend but will be getting started on this now.

The general roadmap for the next weeks looks like this:

I want to bring back the core functionality of the site back online as soon as possible, so we have a good starting point for future features and improvements. One thing I have realized over the past months is that it's probably near-impossible to build up a community of contributors without an existing site and code base to talk about, since issues are not tangible enough and too broad. So my focus for the next weeks will be to bring back a baseline Toolbox that anyone interested can start contributing to.

I wrote above and also for example in #4 what I consider the core features, and from I think this is supported by most comments made in this regard in this issue:

  * Categories
  * An easy way to see key project metrics at a glance (see also #4 and #5 on this)
  * Search

I've decided to, at least for now, go with plain rubygems.org and Github APIs as the data backends since after a deeper look into the libraries.io APIs and available data (see #5) many metrics I consider important are not available in their data set. I still like the idea of acquiring data from libraries.io since this would enable other language communities to build on this code base, too, but like I said I also want to bring back the Toolbox with a decent feature set as soon as possible, and I think only the direct access to rubygems and github can provide that at the moment. Swapping data providers down the road should not be too hard though, so once we're in gentler waters we probably should re-evaluate and sync up with the libraries.io team.

Regarding data, I'll try to go for the low-hanging fruit first, meaning the data that is available directly in the upstream APIs.

Regarding the categorization of projects, the old Toolbox used to have a built-in feature that allowed to send in change requests for projects, i.e. adding a category. To avoid the burden of maintaining this feature in the future and to allow easier community contributions, discussion and reviews I set up a new repository at https://github.com/rubytoolbox/catalog which contains the Toolbox's categories in a straight forward YAML file and directory structure. The repo holds a README containing a description of the format. I'll be working on adding a CI parser for validation and export of this soon.

**If you'd like to help, this is where you can get started right away!** As I've said before, the categories have seen too little maintenance love from me in recent years, so we have a lot of spring cleaning, adding of new projects (and removal of dead one's?) to do, and this can begin right away over at https://github.com/rubytoolbox/catalog

This got longer than expected once again :) I'll try to get better at giving more incremental, short updates, but I felt it neccessary to give everyone an overview of the current status.

Finally, thanks again to everyone for your encouragement, your kind messages, :+1:s and :heart:s mean a lot to me!

Cheers,
Christoph


[tweet-1]: https://twitter.com/TheDeadSerious/status/913448436032208896
[tweet-2]: https://twitter.com/rubytogether/status/913459945881735168

**Comment by arbox** [comment]
Very good news! I'll try to incorporate the data from my lists into the Catalog

- [Ruby NLP](https://github.com/arbox/nlp-with-ruby)
- [Ruby Machine Learning](https://github.com/arbox/machine-learning-with-ruby)
- [Ruby Data Science](https://github.com/arbox/data-science-with-ruby)

@colszowka please look at this issue:  https://github.com/rubytoolbox/rubytoolbox/issues/8

**Comment by chrishough** [comment]
This is great news ‼️ 🎆

**Comment by patcon** [comment]
Yay to sponsorship news! Just wanted to point out that OpenCollective.com might work for you as well -- it's a site that helps unincorporated open source projects handle the tax implications for open source work of certain types -- OpenCollective Foundation is a 503(c)3 that they run to be a "fiscal sponsor" (ie. loan a slice of their bank account) to open source projects.

Might be a way to slowly spin up sustainable funding in recurring donations, for if/when @rubytogether support runs out, or in case you need to bill more hours.

@piamancini can probably clarify if I got any of the facts out of whack :)

**Comment by piamancini** [comment]
@rubytogether is a host on open collective so they could hots this collective. Otherwise, the [Open Source Collective](https://opencollective.com/opensource) would be happy to.
Just to clarify @patcon the 501c3 hasn't been approved yet and in any case, that entity cant' be the fiscal sponsor for open source projects. The Open Source Collective is a 501c6 (the difference is that it's that while it is a non profit, it can't provide tax deductible receipts).

**Comment by indirect** [comment]
For the record, @rubytogether plans to fund this work in a sustainable and ongoing way. The more developers and companies chip in, the more work on Bundler, RubyGems, and Ruby Toolbox we will be able to fund.

And if @colszowka wants to run an Open Collective just for Ruby Toolbox, we would also be happy to host that collective at Ruby Together. 😄

**Comment by patcon** [comment]
Yay sub-collectives!

**Comment by merlin2049er** [comment]
I'd join an open source project.  I need all the experience I can get.

**Comment by pboling** [comment]
@merlin2049er Head to https://github.com/rubytoolbox/, fork a project, check the issue tracker, comment, code, push, repeat.

**Comment by OriPekelman** [comment]
Hi,

Platform.sh will be happy to provide free hosting for this; Just sayin. If interested contact me.

O.

**Comment by colszowka** [comment]
Hey everyone,

the site is back :tada: :grinning: - https://www.ruby-toolbox.com/blog/2018-02-01/lets-push-things-forward

Best,
Christoph

**Comment by joshuapinter** [comment]
![giphy 8](https://user-images.githubusercontent.com/180819/35703896-f2f21c62-075a-11e8-891a-937e639f6461.gif)

**Comment by franklinyu** [comment]
Oh man, the new site looks very sexy…

**Comment by mchilson** [comment]
Awesome work! **Thank you!**

**Comment by chrishough** [comment]
This is awesome news! Super stoked to see this.

**Comment by xecutioner** [comment]
Thank you sir !
